src/apps/replication/lib/mutation_log.cpp

```diff
@@ -1126,10 +1159,11 @@ int mutation_log::garbage_collection(replica_log_info_map& durable_decrees)
                 }
 
                 global_partition_id gpid = kv.first;
-                decree durable_decree = kv.second.max_decree;
+                decree gc_max_decree = kv.second.max_decree;
                 int64_t valid_start_offset = kv.second.valid_start_offset;
 
-                bool delete_ok_for_this_replica;
+                bool delete_ok_for_this_replica = false;
+                bool kickout_this_replica = false;
```

Comment:
I don't get the usage of kickout_this_replica, and its assignments are exactly the same compared to delete_ok_for_this_replica. Any particular reason to introduce this new variable?